### PR TITLE
remove references to binder

### LINKED ISSUE #903: Ensure that notebooks can run in Binder out of the box
(Follow-up from discussion https://github.com/opendp/opendp/discussions/898)

The first cell of `first-look-at-DP.ipynb` failed for me on Binder because pandas is not installed. Should fix it by adding pandas to `.binder/requirements.txt`.

(... but `smoke-test.yml` seems like it should be testing notebooks with `pytest --nbmake source/**/*.ipynb -n=auto`, so maybe that test isn't working as intended?)

I'll try starting a PR, and will update here with what I learn. 

Ideally, close when all notebooks work in Binder if cells are evaluated in order, and CI tests fail if this is not true.

### LINKED ISSUE #955: BinderHub doesn't load for the "Getting started" tutorial
Heya,

I clicked on the "launch binder" button on top of [the Getting Started page](https://docs.opendp.org/en/v0.8.0/user/getting-started.html), but the notebook doesn't load. After 15 or so minutes, I get a big message that says:

```
Error loading opendp/opendp/v0.8.0!
See logs below for details.
Your session is taking longer than usual to start!
Check the log messages below to see what is happening.
```

and here are the logs below:

```
Found built image, launching...
Launching server...
Server requested
2023-10-09T09:37:53.209530Z [Normal] Successfully assigned ovh2/jupyter-opendp-2dopendp-2dfge8upus to user-202211a-node-39c69b
2023-10-09T09:37:54Z [Normal] Container image "jupyterhub/mybinder.org-tc-init:2020.12.4-0.dev.git.4289.h140cef52" already present on machine
2023-10-09T09:37:54Z [Normal] Created container tc-init
2023-10-09T09:37:54Z [Normal] Started container tc-init
2023-10-09T09:37:55Z [Normal] Pulling image "2lmrrh8f.gra7.container-registry.ovh.net/mybinder-builds/r2d-g5b5b759opendp-2dopendp-b2d0b1:13cf95652a95545f5250324e6bee91a57df4a5cd"
Launch attempt 1 failed, retrying...
Server requested
2023-10-09T09:40:32.708146Z [Normal] Successfully assigned ovh2/jupyter-opendp-2dopendp-2dtawue1az to user-202211a-node-25cdfe
2023-10-09T09:40:33Z [Normal] Container image "jupyterhub/mybinder.org-tc-init:2020.12.4-0.dev.git.4289.h140cef52" already present on machine
2023-10-09T09:40:33Z [Normal] Created container tc-init
2023-10-09T09:40:33Z [Normal] Started container tc-init
2023-10-09T09:40:35Z [Normal] Pulling image "2lmrrh8f.gra7.container-registry.ovh.net/mybinder-builds/r2d-g5b5b759opendp-2dopendp-b2d0b1:13cf95652a95545f5250324e6bee91a57df4a5cd"
Spawn failed: Timeout
Launch attempt 2 failed, retrying...
Launch attempt 3 failed, retrying...
Server requested
2023-10-09T09:46:37.177325Z [Normal] Successfully assigned ovh2/jupyter-opendp-2dopendp-2d5e3zt4jn to user-202211a-node-25cdfe
2023-10-09T09:46:38Z [Normal] Container image "jupyterhub/mybinder.org-tc-init:2020.12.4-0.dev.git.4289.h140cef52" already present on machine
2023-10-09T09:46:38Z [Normal] Created container tc-init
2023-10-09T09:46:38Z [Normal] Started container tc-init
2023-10-09T09:46:39Z [Normal] Pulling image "2lmrrh8f.gra7.container-registry.ovh.net/mybinder-builds/r2d-g5b5b759opendp-2dopendp-b2d0b1:13cf95652a95545f5250324e6bee91a57df4a5cd"
Image 2lmrrh8f.gra7.container-registry.ovh.net/mybinder-builds/r2d-g5b5b759opendp-2dopendp-b2d0b1:13cf95652a95545f5250324e6bee91a57df4a5cd for user opendp-opendp-5e3zt4jn failed to launch
```

### LINKED ISSUE #1873: Remove the binder link, to make sure no one will miss it
Long term plan is to move from notebooks to RST. (#1667, for example.) One thing we'll lose with this is the launch-in-binder. (Unless we also are planning to generate notebooks from RST? Not impossible, but a big job.)

One thing we could do to test the waters is remove the binder link now... and if this turns out to cause trouble, we can bring the link back and reassess our plans with respect to RST.

### PR DESCRIPTION
**Pull request by mccalluc**
- Fix #1873
- also, fix #903, or at least render it moot
- close #955, by dropping binder

If it turns out we _don't_ want to do this, that would also be a good thing to learn.

### CODE REVIEW

File: tools/assert_version.py [hunk 1]
```diff
@@ -31,12 +31,6 @@
 assert config['metadata']['version'] == str(python_version), \
     "python/setup.cfg package version is incorrect"
 
-binder_requirements = open('.binder/requirements.txt').readlines()
-for line in binder_requirements:
-    if line.startswith("opendp=="):
-        assert line == f"opendp=={python_version}\n", \
-            ".binder/requirements.txt opendp dependency is incorrect"
```
Comment (mccalluc) [inline tools/assert_version.py:38]: If this PR doesn't move forward, should file an issue to make a fix here: `opendp[polars]` is now specified, so the match wouldn't catch what we expect it to.

File: tools/channel_tool.py [hunk 2]
```diff
@@ -114,19 +114,12 @@ def munge_python_config(config):
     def dump_python_config(config, f):
         config.write(f)
     update_file("python/setup.cfg", load_python_config, munge_python_config, dump_python_config)
-
-    # Binder requirements
-    def munge_binder_requirements(lines):
-        opendp_line = f"opendp=={python_version}\n"
-        return [opendp_line if line.startswith("opendp==") else line for line in lines]
```
Comment (mccalluc) [inline tools/channel_tool.py:121]: ditto: if PR doesn't move forward, fix this.

**Review by mccalluc — COMMENTED** [review]
Looking at the current state, before this PR, there are some bugs because of changes to `.binder/requirements.txt` that should be addressed even if this PR isn't the right direction.

**Review by Shoeboxam — APPROVED** [review]
Binder makes so much sense in theory, but the slow startup times and outages have made it un-economical. We're also gradually moving away from notebooks to better support multi-language documentation.

**Pull request merged**